File: app/src/main/java/com/inspiringteam/reactivecompass/utils/FormatUtils.java
package com.inspiringteam.reactivecompass.utils;

import java.text.DecimalFormat;
import java.text.NumberFormat;

public class FormatUtils {
    public static String getStringFromDouble(double d) {
        NumberFormat formatter = new DecimalFormat("###.#######");
        return formatter.format(d);
    }
}

File: app/src/main/java/com/inspiringteam/reactivecompass/utils/FormatUtils.kt
package com.inspiringteam.reactivecompass.utils

import java.text.DecimalFormat
import java.text.NumberFormat

object FormatUtils {
    fun getStringFromDouble(d: Double): String {
        val formatter = DecimalFormat("###.#######")
        return formatter.format(d)
    }
}
